ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -292,4 +338,66 @@ protected String getReplicaThreadName(String datacenterToReplicateFrom, int thre
   protected boolean shouldReplicateFromDc(String datacenterName) {
     return cloudConfig.vcrSourceDatacenters.contains(datacenterName);
   }
+
+  /**
+   * Check and schedule a VCR helix update task. We schedule updateVcrHelix in 2 cases:
+   * 1. On a node become online role of partition 1
+   * 2. On Ambry cluster change
+   */
+  private void scheduleVcrHelix() {
+    if (vcrHelixUpdateFuture != null && vcrHelixUpdateFuture.cancel(false)) {
+      // If a vcrHelixUpdate task is scheduled, try to cancel it first.
+      logger.info("There was a scheduled vcrHelixUpdate task. Canceled.");
+    }
+    // either success cancel or not, we should schedule a new job to updateVcrHelix
+    vcrHelixUpdateFuture =
+        scheduler.schedule(() -> updateVcrHelix(), cloudConfig.vcrHelixUpdateDelayTimeInSeconds, TimeUnit.SECONDS);
```

Comment:
Why do we delay the `updateVcrHelix` execution? Is it to make sure that any previously scheduled update is cancelled before starting the task? If yes, should we rather try to wait for that update to be cancelled?

Reply:
If we add multiple partitions, VCR may receive multiple updates, but we only need to update it once. 
So we delay the update for a short time window, if there is another update comes in the short time window, we just cancel the old pending update.

Reply:
The delay time should be greater than the time to upgrade an Ambry cluster. With that we will have a better chance that we have taken in all the cluster updates, before we go and update the vcr helix cluster.

Reply:
A longer delay window is preferred, but I don't understand why it should > time to upgrade vcr helix cluster.